polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -633,10 +661,26 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				let assets = Self::reanchored(deposited, &dest, None);
 				let mut message = vec![ReserveAssetDeposited(assets), ClearOrigin];
 				message.extend(xcm.0.into_iter());
+
+				// put back parked_fee in holding register to be charged by XcmSender
+				self.holding.subsume_assets(parked_fee);
 				self.send(dest, Xcm(message), FeeReason::DepositReserveAsset)?;
 				Ok(())
 			},
 			InitiateReserveWithdraw { assets, reserve, xcm } => {
+				// we need to do this take/put cycle to solve wildcards and get exact assets to be
```

Comment:
This is less of a problem since it happens on the origin chain, and not the intermediary chain. We could thus use JIT withdrawal to resolve this.

However, we may want a consistent UX, which would require us to not use JIT withdrawal and remove the SetFeesMode instruction when executing the initial teleport or reserve asset transfer.

Reply:
The thing with jit withdrawal is that it's not expected.
Users pay `assets` and expect execution fees to be deducted from that amount.
With jit withdrawal we are deducting delivery fees from the user's account itself, which might not have more assets after the transfer.
So user has to make sure they pay `assets` and also have enough for delivery fees.
This makes it so they only pay `assets` and both execution and delivery fees are deducted from that amount.

Reply:
this is definitely needed for InitiateReserveWithdraw too but not for the current scenario where the intermediary chain is the reserve - I say we keep it, but add a check if jit_withdraw is set - if it is, then we can skip it and let it withdraw from origin's account instead of holding

(maybe we do this jit check for all of them, but honestly I would just deprecate JIT_WITHDRAW altogether, it just introduces extra complexity and error cases)

Reply:
I removed the `SetFeesMode { jit_withdraw: true }` instruction from the teleport and reserve asset transfer extrinsics and re-run the benchmarks. It should only charge for the fees once now and in a more consistent way.

Reply:
that's true for `pallet-xcm`, but what about other xcm pallets (e.g. `xtokens` who surfaced this)?

we can't assume in executor that no one will use `SetFeesMode { jit_withdraw: true }` - as long as we have it, we have to assume it can/will be used

Reply:
@franciscoaguirre, just to confirm; this means that we can't just send now (as an example) USDT to a para chain from AssetHub. We also have to include KSM and AT LEAST the amount of the XCM delivery fee? And then whatever we put over the min XCM Delivery Fee amount will be delivered to the target account in the target chain?